src/main/java/com/google/sps/data/LocationData.java

```diff
@@ -42,6 +43,22 @@ public LocationData(EmbeddedEntity embeddedLocation) {
     this.radius = ((Double) embeddedLocation.getProperty("radius")).floatValue();
   }
 
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
Nit: Shouldn't this be an Object* (pointer to an object) if you are going to compare to |this|?

Reply:
Java doesn't really use pointers in that way. But equals and hashCode are often boiler-plate code that can be generated automagically by your IDE. Not sure if this was done by hand or with a code generation tool?

nit: in my experiences, since equals and hashCode are often generated, and rarely called directly, they are typically stuck at the bottom of the class, after all the getters/setters.

Reply:
It was done by hand 😅 , I did research about it to see if there was like a common implementation.
And yep, because of how Java works, when you compare `this == obj`, you are comparing that their references point to the same object.